### PR TITLE
Update pci.ids file and update logic to extract the device name

### PR DESCRIPTION
**Pull request by cdesiniotis**
Depends on https://github.com/NVIDIA/kubevirt-gpu-device-plugin/pull/71

cc @rthallisey @zvonkok

### CODE REVIEW

**Comment by rthallisey** [comment]
@cdesiniotis can you rebase?

**Comment by cdesiniotis** [comment]
Rebased.

File: pkg/device_plugin/device_plugin.go [hunk 1]
```diff
@@ -308,32 +309,46 @@ func getDeviceName(deviceID string) string {
...
-			}
-			deviceName = strings.TrimSpace(splits[1])
-			deviceName = strings.ToUpper(deviceName)
-			deviceName = strings.Replace(deviceName, "/", "_", -1)
-			deviceName = strings.Replace(deviceName, ".", "_", -1)
-			reg, _ := regexp.Compile("\\s+")
-			deviceName = reg.ReplaceAllString(deviceName, "_") // Replace all spaces with underscore
-			reg, _ = regexp.Compile("[^a-zA-Z0-9_.]+")
-			deviceName = reg.ReplaceAllString(deviceName, "") // Removes any char other than alphanumeric and underscore
+		if strings.HasPrefix(line, "10de") {
```
Comment (rthallisey) [inline pkg/device_plugin/device_plugin.go:316]: Why are there two `scanner.Scan()` calls?  Seems like the `for scanner.Scan() { .. ` code below does what you need.

Comment (cdesiniotis) [reply pkg/device_plugin/device_plugin.go:316]: Without the first for loop, we are only considering the device id, not vendor id.

Consider the below example (which exists now in the updated pci.ids file):
```
8086  Intel Corporation
	2331  DH89xxCC Chap Counter
10de NVIDIA Corporation
	2331 GH100 [H100 PCIe]
```
The Intel device shares the same device id as H100 PCIe. The first for loop ensures we only start looking for a matching device id once we have reached the section containing NVIDIA devices.

Comment (rthallisey) [reply pkg/device_plugin/device_plugin.go:316]: Ok I would put this behind a function then so it reads a bit better.

```go
const nvidiaVendorID = 10de
locateVendor(nvidiaVendorID)

for scanner.Scan() { ... }
```

Comment (cdesiniotis) [reply pkg/device_plugin/device_plugin.go]: Updated as you have suggested.

**Comment by rthallisey** [comment]
lgtm

**Comment by rthallisey** [comment]
Thx Chris

**Pull request merged**